src/middleware.ts

```diff
@@ -29,6 +29,14 @@ export function middleware(request: NextRequest) {
     existingExperimentationId?.value ?? `guest-${crypto.randomUUID()}`;
   requestHeaders.set("x-experimentation-id", experimentationId);
 
+  // Check for Nimbus preview mode. Note that this requires a full page reload
+  // to activate: https://nextjs.org/docs/app/api-reference/file-conventions/layout#caveats
+  const nimbusPreviewMode = request.nextUrl.searchParams.get("nimbus_preview");
+  requestHeaders.set(
+    "x-nimbus-preview-mode",
+    nimbusPreviewMode === "true" ? "true" : "false",
```

Comment:
Nit: This could be simplified to:
```suggestion
    (nimbusPreviewMode === "true").toString(),
```

Reply:
Thinking about it again: The current ternary operator might be easier to read.

Reply:
I took a look at the existing code and I see some examples of doing this same thing as ternary but not `toString()`. I like the latter method since it's a little shorter (and arguably clearer, I think it'd be easy to gloss over the fact that the ternary is *also* returning a string) but I'd rather be consistent for now and switch everything else over too if we're going to go that way. Good suggestion :)

Reply:
Oh also I don't like that the ternary could easily typo e.g. `treu` or `fales` which to `toString()` approach doesn't have.